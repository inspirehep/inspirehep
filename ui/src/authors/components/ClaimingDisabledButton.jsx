import React from 'react';
import { FileDoneOutlined } from '@ant-design/icons';
import { Button, Tooltip } from 'antd';

import DropdownMenu from '../../common/components/DropdownMenu';
import IconText from '../../common/components/IconText';
import ListItemAction from '../../common/components/ListItemAction';

const ClaimingDisabledButton = () => <ListItemAction>
  <DropdownMenu
    disabled
    title={
      <Tooltip title='Login to claim your papers'>
        <Button disabled>
          <IconText text="claim" icon={<FileDoneOutlined />} />
        </Button>
      </Tooltip>
    }
  />
</ListItemAction>;

export default ClaimingDisabledButton;
