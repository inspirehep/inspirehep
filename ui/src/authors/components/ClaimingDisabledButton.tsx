import React from 'react';
import { FileDoneOutlined } from '@ant-design/icons';
import { Button, Tooltip } from 'antd';

import DropdownMenu from '../../common/components/DropdownMenu';
import IconText from '../../common/components/IconText';
import UserAction from '../../common/components/UserAction';

const ClaimingDisabledButton = () => <UserAction>
  <DropdownMenu
    disabled
    title={
      <Tooltip title='Login to claim your papers'>
        <Button disabled>
          <IconText text="claim" icon={<FileDoneOutlined />} />
        </Button>
      </Tooltip>
    }
  />
</UserAction>;

export default ClaimingDisabledButton;
