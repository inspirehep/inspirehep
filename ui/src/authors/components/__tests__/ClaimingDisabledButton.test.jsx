import React from 'react';
import { render } from '@testing-library/react';

import ClaimingDisabledButton from '../ClaimingDisabledButton';

describe('ClaimingDisabledButton', () => {
  it('renders', () => {
    const { asFragment } = render(<ClaimingDisabledButton />);
    expect(asFragment()).toMatchSnapshot();
  });
});
