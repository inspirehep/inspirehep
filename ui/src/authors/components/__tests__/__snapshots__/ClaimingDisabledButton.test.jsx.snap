// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimingDisabledButton renders 1`] = `
<ListItemAction>
  <DropdownMenu
    disabled={true}
    title={
      <ForwardRef
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement="top"
        title="Login to claim your papers"
      >
        <ForwardRef(InternalButton)
          disabled={true}
        >
          <IconText
            icon={<ForwardRef(FileDoneOutlined) />}
            text="claim"
          />
        </ForwardRef(InternalButton)>
      </ForwardRef>
    }
  />
</ListItemAction>
`;
